src/rest_framework_dso/pagination.py

```diff
@@ -137,10 +133,35 @@ def __init__(self, results_field=None):
     def get_paginated_response(self, data):
         """Wrap the data in all pagination parts."""
         # Add the _links, and add the HTTP headers.
+        self.request.accepted_renderer.setup_pagination(self)
         data = self._get_paginated_data(data)
         return super().get_paginated_response(data)
 
     def _get_paginated_data(self, data: ReturnList) -> dict:
+        if isinstance(data, dict):
```

Comment:
oeps.. de `data: ReturnList` annotatie hierboven klopt dus niet... Is denk ik `Union[ReturnList, ReturnDict]` en wellicht zelfs `ReturnGenerator`.